packages/gotrue/lib/src/fetch.dart

```diff
@@ -50,24 +63,44 @@ class GotrueFetch {
           message: error.toString(), originalError: error);
     }
 
-    // Check if weak password reasons only contain strings
-    if (data is Map &&
-        data['weak_password'] is Map &&
-        data['weak_password']['reasons'] is List &&
-        (data['weak_password']['reasons'] as List).isNotEmpty &&
-        (data['weak_password']['reasons'] as List)
-            .whereNot((element) => element is String)
-            .isEmpty) {
+    String? errorCode;
+
+    final responseApiVersion = ApiVersion.fromResponse(error);
+
+    if (responseApiVersion?.isSameOrAfter(ApiVersions.v20240101) ?? false) {
+      errorCode = _getErrorCode(data, 'code');
+    } else {
+      errorCode = _getErrorCode(data, 'error_code');
+    }
```

Comment:
This could still fail somehow, specially because the version is set in the client if it is missing in the response headers. What about something like:

```suggestion
    String? errorCode = _getErrorCode(data, 'code') ?? _getErrorCode(data, 'error_code');
```

Reply:
Hmm, in what scenario would the version in the response headers be missing though, and how would it "fail"? In that case, wouldn't it just go to `else` in the above code block returning `_getErrorCode(data, 'error_code')`?

Reply:
The gotrue library is not restricted to Supabase, and it is open source and its source can be changed.

Yes, it fallbacks to the else clause, but it would imply that the first clauze isn't tested in some case.

But we can keep the version check, it is fine for me.

Reply:
Thanks for the catch! I think at this point we can consider this library dedicated to Supabase, so let's go with the current code 👍